rows/utils.py

```diff
@@ -297,3 +310,53 @@ def export_to_uri(table, uri, *args, **kwargs):
         raise ValueError('Plugin (export) "{}" not found'.format(plugin_name))
 
     return export_function(table, uri, *args, **kwargs)
+
+
+def decompress(path, **kwargs):
+    """
+    Given a bz2, gzip or lzma file returns a decompressed file object. All
+    kwargs are passed to either `bz2.openn`, `gzip.open` or `lzma.open`.
+    :param path: (str) path to a bz2, gzip or lzma file
+    """
+    filename = os.path.basename(path)
+    with open(path, 'rb') as handler:
+        mime_type = describe_file_type(filename, handler.read())[0]
+
+    bz2_mime_types = (
+        'application/bzip2',
+        'application/octet-stream',
+        'application/x-bz2',
+        'application/x-bzip',
+        'application/x-compressed',
+        'application/x-stuffit'
+    )
+    gzip_mime_types = (
+        'application/gzip',
+        'application/x-gzip',
+        'application/x-gunzip',
+        'application/gzipped',
+        'application/gzip-compressed',
+        'application/x-compressed',
+        'application/x-compress',
+        'gzip/document',
+        'application/octet-stream'
+    )
+    lzma_mime_types = (
+        'application/x-xz',
+        'application/x-lzma'
+    )
```

Comment:
I think mimetype detection is not needed here - we can expect the user will call this function only if she knows the file is compressed and in one of the supported algorithms; we can do this detection automatically on the command-line interface using [file-magic](http://pypi.python.org/pypi/file-magic) and then pass the correct arguments to `decompress`. Do you agree?

Reply:
Totally : )